### PR TITLE
Migrate to using a new Exchange contract

### PR DESCRIPTION
**Pull request by LogvinovLeon**
This PR:
* Updates node from 0.6.1 to 0.6.5
* Uses yarn on CI
* Simplifies BigNumber hack
* Is essentially a bunch of renames and a copy of contract artifacts.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,16 +1,23 @@
 machine:
   node:
-    version: 6.1.0
+    version: 6.5.0
   environment:
-      CONTRACTS_COMMIT_HASH: cd04d7c
+      CONTRACTS_COMMIT_HASH: '414e997'
+      PATH: "${PATH}:${HOME}/${CIRCLE_PROJECT_REPONAME}/node_modules/.bin"
```
Comment (fabioberger) [inline circle.yml]: Where is `CIRCLE_PROJECT_REPONAME ` defined? Does CI set this env?

File: src/contract_wrappers/exchange_wrapper.ts [hunk 2]
```diff
@@ -198,16 +201,16 @@ export class ExchangeWrapper extends ContractWrapper {
...
     }
     /**
      * Sequentially and atomically fills signedOrders up to the specified takerTokenFillAmount.
      * If the fill amount is reached - it succeeds and does not fill the rest of the orders.
      * If fill amount is not reached - it fills as much of the fill amount as possible and succeeds.
      * @param   signedOrders            The array of signedOrders that you would like to fill until
      *                                  takerTokenFillAmount is reached.
-     * @param   takerTokenFillAmount    The total amount of the takerTokens you would like to fill.
+     * @param   fillTakerTokenAmount    The total amount of the takerTokens you would like to fill.
      * @param   shouldCheckTransfer     Whether or not you wish for the contract call to throw if upon
```
Comment (fabioberger) [inline src/contract_wrappers/exchange_wrapper.ts]: Forgot to rename comment param to `shouldThrowOnInsufficientBalanceOrAllowance `.

File: src/contract_wrappers/exchange_wrapper.ts [hunk 3]
```diff
@@ -198,16 +201,16 @@ export class ExchangeWrapper extends ContractWrapper {
...
      *                                  execution any of the tokens cannot be transferred. If set to false,
      *                                  the call will continue to fill subsequent signedOrders even when
```
Comment (fabioberger) [inline src/contract_wrappers/exchange_wrapper.ts]: Missing `takerAddress` param comment.

File: src/contract_wrappers/exchange_wrapper.ts [hunk 4]
```diff
@@ -482,36 +491,36 @@ export class ExchangeWrapper extends ContractWrapper {
      */
     @decorators.contractCallErrorHandler
     public async cancelOrderAsync(
-        order: Order|SignedOrder, takerTokenCancelAmount: BigNumber.BigNumber): Promise<BigNumber.BigNumber> {
+        order: Order|SignedOrder, canceltakerTokenAmount: BigNumber.BigNumber): Promise<BigNumber.BigNumber> {
```
Comment (fabioberger) [inline src/contract_wrappers/exchange_wrapper.ts]: Need to update @param comment name to `canceltakerTokenAmount`.

Comment (fabioberger) [inline src/contract_wrappers/exchange_wrapper.ts]: Also, I know there was a typo in the smart contract, but we shouldn't propagate that error to 0x.js. Here, we should use the appropriate camel-casing: `cancelTakerTokenAmount`.

File: circle.yml (hunk 1 above)
Comment (LogvinovLeon) [inline circle.yml]: Yes

**Pull request merged**